### DISCUSSION TITLE
Center / align matplotlib figures

### DISCUSSION THREAD

**Discussion by lu-kas** [executablebooks/meta#341] [Q&A]
Hi!

Is it possible to center the figures generated by a code cell in the output field? I have tried to add a custom CSS style, but was not successful -- probably because I'm not very familiar with CSS.

Does anyone have an idea? 

Thanks!

<img width="831" alt="image" src="https://user-images.githubusercontent.com/8122581/116774877-dfe6ec00-aa5f-11eb-8f65-c9bbc94de8a4.png">

**Comment by choldgraf** [comment]
Could you try the instructions here? "Formatting code outputs" https://jupyterbook.org/content/code-outputs.html#images

**Reply by lu-kas** [reply]
Thanks for the prompt response. I was not clear enough in my question, sorry.

Background: I work with jupyter notebooks and build the book out of them. Thus, I can share the code / methods / etc. with my students and they can 'play' with them. For me, it is more preferable to work with jupyter, because this way I get an IDE and can directly see and test things -- so having a pure MD file is very unpractical for me.

Regarding your answer: As far as I understand the info in 'Formatting code outputs', this is only true for MD files and thus ignored this piece of information. Nevertheless, I tried it in jupyter and it results in

` WARNING: Found an unexpected `code-cell` directive. Either this file was not converted to a notebook, because Jupytext header content was missing, or the `code-cell` was not converted, because it is nested. See https://myst-nb.readthedocs.io/en/latest/use/markdown.html for more information.`

and does not produce any output (in the book).

Rephrased question: How can I align, e.g., matplotlib figures within the jupyterbook page of a jupyter notebook?  

Thanks for your support and for the great project!

**Reply by lu-kas** [reply]
Ok, I have done some digging. Eventually, the perfect solution for me would be to directly generate a myst MD cell with the matplotlib generated image as a figure in the book. This way I could preserve my jupyter workflow. 

Thus, I have written a function, which grabs the matplotlib figure object, saves the plot to a figure and generates the according MD output. It works in jupyter:

<img width="843" alt="image" src="https://user-images.githubusercontent.com/8122581/116814672-64b63080-ab5a-11eb-8d9c-035188a06ad1.png">

But does not get properly interpreted in jupyterbook:

<img width="828" alt="image" src="https://user-images.githubusercontent.com/8122581/116814697-844d5900-ab5a-11eb-9b1a-e1021d4a6fb2.png">

Any idea how to trigger the MD cell evaluation?

This may be related to the discussion further down in https://github.com/executablebooks/jupyter-book/issues/1105?

Or, this one https://github.com/executablebooks/jupyter-book/issues/1069?

Thanks!
